### ISSUE TITLE
[New rule] Widgets to use Theme

### ISSUE THREAD

**Issue by NickBalnaves** [dart-code-checker/dart-code-metrics#745]
**Please describe what the rule should do:**
Any text styling or colors used in widgets to come from a Theme so these values can be shared throughout an app.

**What category of rule is this? (place an "X" next to just one item)**
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**
Good:
```dart
Container(
  color: Theme.of(context).colorScheme.secondary,
  child: Text(
    'Text with a background color',
    style: Theme.of(context).textTheme.headline6,
  ),
)
```

Bad:
```dart
Container(
  color: Colors.red,
  child: Text(
    'Text with a background color',
    style: TextStyle(color: Color(0xFF32DD35), fontSize: 14),
  ),
)
```

**Are you willing to submit a pull request to implement this rule?**
No

**Comment by incendial** [comment]
@NickBalnaves hi, do you have different examples of getting the theme? I mean, the rule should probably trigger on something like:
```
final theme = Theme.of(context);
...
Container(
  color: theme.colorScheme.secondary,
  child: Text(
    'Text with a background color',
    style: theme.textTheme.headline6,
  ),
)
```

but are there more ways to get the theme?

**Comment by NickBalnaves** [comment]
Hi @incendial, here's another example which should cause a lint but seems that it might be more difficult to catch:
```dart
ExampleWidget(Theme.of(context));

class ExampleWidget extends StatelessWidget {
  const ExampleWidget(final this.firstTheme);
  final ThemeData firstTheme;

  @override
  Widget build(BuildContext context) {
    final secondTheme = Theme.of(context);
    ...
  }
}
```

It's possible to define a new class to use for a theme instead of the default material Theme, something like this:
```dart
class CustomTheme extends StatelessWidget {
  ...
  static CustomThemeData of(BuildContext context) {
    ...
```
so it might be good to also add an override to define a custom theme.

**Comment by incendial** [comment]
Thanks, got it. Right now looks hard to cover all the edge cases, but we can probably start with the partial support.
